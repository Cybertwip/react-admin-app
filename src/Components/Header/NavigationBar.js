import React from 'react';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink } from 'reactstrap';

export default class NavigationBar extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      isOpen: false
    };
  }
  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }
  render() {
    return (
      <div>
        <Navbar color="light" light expand="md">
          <NavbarBrand href="/">Web Admin Panel</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-auto" navbar>
            <NavItem>
                <NavLink href="/crud">CRUD</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/resources">Resources</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="https://github.com/VictorJL/react-web-admin-app" target="_blank">GitHub</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}
