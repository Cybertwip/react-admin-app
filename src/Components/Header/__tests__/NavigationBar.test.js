import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

import renderer from 'react-test-renderer';

import NavigationBar from './../NavigationBar';

describe('Main component rendering', () => {
    it('Full-render renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<NavigationBar />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Shallow-render renders without crashing', () => {
      shallow(<NavigationBar />);
    });

    it('Render to match snapshot', () => { // 1
      const tree = renderer.create(<NavigationBar />).toJSON();
      
      expect(tree).toMatchSnapshot();
    });
});
