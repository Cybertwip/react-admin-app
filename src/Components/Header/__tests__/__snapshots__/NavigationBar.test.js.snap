// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component rendering Render to match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Web Admin Panel
    </a>
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/crud"
            onClick={[Function]}
          >
            CRUD
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/resources"
            onClick={[Function]}
          >
            Resources
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://github.com/VictorJL/react-web-admin-app"
            onClick={[Function]}
            target="_blank"
          >
            GitHub
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
